diff --git a/node_modules/source-map-support/source-map-support.js b/node_modules/source-map-support/source-map-support.js
index 4459386..09ff435 100644
--- a/node_modules/source-map-support/source-map-support.js
+++ b/node_modules/source-map-support/source-map-support.js
@@ -476,7 +476,6 @@ function printErrorAndExit (error) {
   }
 
   console.error(error.stack);
-  process.exit(1);
 }
 
 function shimEmitUncaughtException () {
